projects/core/src/main/resources/data/computercraft/lua/rom/modules/main/cc/audio/wav.lua

```diff
@@ -0,0 +1,224 @@
+                            c[j] = chunk[(j - 1) * channels + i]
+                        end
+                    end
+                    return table.unpack(res)
+                end
+            else
+                local format = dataType == "unsigned" and "I" .. (bitDepth / 8) or (dataType == "signed" and "i" .. (bitDepth / 8) or "f")
+                local transform
+                if dataType == "unsigned" then
+                    function transform(n) return n - 128 end
+                elseif dataType == "signed" then
+                    if bitDepth == 16 then function transform(n) return math_floor(n / 0x100) end
+                    elseif bitDepth == 24 then function transform(n) return math_floor(n / 0x10000) end
+                    elseif bitDepth == 32 then function transform(n) return math_floor(n / 0x1000000) end end
+                elseif dataType == "float" then
+                    function transform(n) return math_floor(n * (n < 0 and 128 or 127)) end
+                end
+                function retval.read(samples)
+                    if pos > #data then return nil end
+                    local chunk = { ("<" .. format:rep(math.min(samples * channels, (#data - pos + 1) / (bitDepth / 8)))):unpack(data, pos) }
+                    pos = table.remove(chunk)
+                    local res = {}
+                    for i = 1, channels do
+                        local c = {}
+                        res[i] = c
+                        for j = 1, samples do
+                            c[j] = transform(chunk[(j - 1) * channels + i])
+                        end
+                    end
+                    return table.unpack(res)
+                end
+            end
+        elseif temp == "fact" then
+            retval.length, pos = str_unpack("<I4", data, pos)
+        elseif temp == "LIST" then
+            local type = str_unpack("c4", data, pos)
+            if type == "INFO" then
+                local e = pos + size
+                pos = pos + 4
+                while pos < e do
+                    local str
+                    type, str, pos = str_unpack("!2<c4s4Xh", data, pos)
+                    str = str:gsub("\0+$", "")
+                    if wavMetadata[type] then retval.metadata[wavMetadata[type]] = tonumber(str) or str
+                    else retval.metadata[type] = tonumber(str) or str end
+                end
+            else pos = pos + size end
+        else pos = pos + size end
+    end
+    if not retval.read then error("invalid WAV file", 2) end
+    return retval
+end
+
+--- Reads a WAV file from a path.
+--
+-- This functions identically to [`readWAV`], but reads from a file instead.
+--
+-- @tparam string path The (absolute) path to read from.
+-- @treturn table The decoded WAV file table.
+-- @see readWAV To read WAV data from a string.
+local function readWAVFile(path)
+    expect(1, path, "string")
+    local file = assert(fs.open(path, "rb"))
+    local data = file.readAll()
```

Comment:
It would be nice if we could avoid reading the whole file into memory at once. I wonder if `readWAV` could take a "reader" function (like `function(bytes: number): string|nil`), so you can just do `readWAV(file.read)` directly.

This would require some changes to parsing the `INFO` section (which uses `string.unpack("s")`), but I think otherwise should be possible?

I guess this would then mean there needs to be a `close` method on the WAV handle, to clean up the underlying file handle. Ughr.